src/formats/png.zig

```diff
@@ -487,101 +56,16 @@ pub const PNG = struct {
-                    4 => PixelFormat.grayscale4,
-                    8 => PixelFormat.grayscale8,
-                    16 => PixelFormat.grayscale16,
-                    else => return ImageError.Unsupported,
-                };
-            },
-            .truecolor => {
-                return switch (self.header.bit_depth) {
-                    8 => PixelFormat.rgb24,
-                    16 => PixelFormat.rgb48,
-                    else => return ImageError.Unsupported,
-                };
-            },
-            .indexed => {
-                return switch (self.header.bit_depth) {
-                    1 => PixelFormat.indexed1,
-                    2 => PixelFormat.indexed2,
-                    4 => PixelFormat.indexed4,
-                    8 => PixelFormat.indexed8,
-                    else => return ImageError.Unsupported,
-                };
-            },
-            .grayscale_alpha => {
-                return switch (self.header.bit_depth) {
-                    8 => PixelFormat.grayscale8Alpha,
-                    16 => PixelFormat.grayscale16Alpha,
-                    else => return ImageError.Unsupported,
-                };
-            },
-            .truecolor_alpha => {
-                return switch (self.header.bit_depth) {
-                    8 => PixelFormat.rgba32,
-                    16 => PixelFormat.rgba64,
-                    else => return ImageError.Unsupported,
-                };
-            },
-        }
-    }
-
-    pub fn findFirstChunk(self: Self, chunk_type: []const u8) ?ChunkVariant {
-        const chunk_id = utils.toMagicNumberBig(chunk_type);
-
-        for (self.chunks.items) |chunk| {
-            if (chunk.getChunkID() == chunk_id) {
-                return chunk;
-            }
-        }
-
-        return null;
-    }
-
-    pub fn getPalette(self: Self) ?PLTE {
-        const palette_variant_opt = self.findFirstChunk(PLTE.ChunkType);
+        var magic_buffer: [types.magic_header.len]u8 = undefined;
 
-        if (palette_variant_opt) |variant| {
-            return variant.PLTE;
-        }
+        _ = try stream.reader().readAll(magic_buffer[0..]);
 
-        return null;
-    }
-
-    pub fn getBackgroundColorChunk(self: Self) ?bKGD {
-        const bkgd_variant_opt = self.findFirstChunk(bKGD.ChunkType);
-
-        if (bkgd_variant_opt) |variant| {
-            return variant.bKGD;
-        }
-
-        return null;
+        return std.mem.eql(u8, magic_buffer[0..], types.magic_header[0..]);
     }
 
     pub fn readImage(allocator: Allocator, stream: *Image.Stream) ImageReadError!Image {
-        var result = Image.init(allocator);
-        errdefer result.deinit();
-        var png = PNG.init(allocator);
-        defer png.deinit();
-
-        try png.read(stream, &result.pixels);
-
-        result.width = png.header.width;
-        result.height = png.header.height;
-
-        return result;
+        var def_options = DefOptions{};
```

Comment:
what's Def? Default? This is not a clear name, use `PngDefaultOptions` or something more clear.